libs/agno/agno/os/app.py

```diff
@@ -318,6 +328,51 @@ def _add_built_in_routes(self, app: FastAPI) -> None:
             self.interfaces.append(a2a_interface)
             self._add_router(app, a2a_interface.get_router())
 
+    def _check_duplicate_ids(
+        self,
+        items: Optional[List[Any]],
+        item_type: str,
+    ) -> None:
+        """Check for duplicate IDs in a list of items.
+
+        Args:
+            items: List of agents, teams, or workflows
+            item_type: Type name for error messages (e.g., "Agent", "Team", "Workflow")
+
+        Raises:
+            ValueError: If duplicate IDs are found
+        """
+        if not items:
+            return
+
+        seen_ids: Dict[str, List[str]] = {}  # id -> list of names with that id
```

Comment:
Why a dict? Rather just add the ID to a list, and if you find it in the list again when you want to add, you have a duplicate.

Reply:
I'll simplify the logic here